extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/discovery/AstroDiscoveryService.java

```diff
@@ -31,39 +32,69 @@
 public class AstroDiscoveryService extends AbstractDiscoveryService {
     private final Logger logger = LoggerFactory.getLogger(AstroDiscoveryService.class);
     private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
     private LocationProvider locationProvider;
+    private ScheduledFuture<?> astroDiscoveryJob;
+    private PointType previousLocation;
+
+    private static ThingUID SUN_THING = new ThingUID(THING_TYPE_SUN, LOCAL);
+    private static ThingUID MOON_THING = new ThingUID(THING_TYPE_MOON, LOCAL);
 
     /**
-     * Creates a AstroDiscoveryService with disabled autostart.
+     * Creates a AstroDiscoveryService with enabled autostart.
      */
     public AstroDiscoveryService() {
-        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, false);
+        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, true);
     }
 
     @Override
     protected void startScan() {
         logger.debug("Starting Astro discovery scan");
-
         PointType location = locationProvider.getLocation();
-
         if (location == null) {
             logger.debug("LocationProvider.getLocation() is not set -> Will not provide any discovery results");
             return;
         }
+        createResults(location);
+    }
 
-        ThingUID sunThing = new ThingUID(AstroBindingConstants.THING_TYPE_SUN, LOCAL);
-        ThingUID moonThing = new ThingUID(AstroBindingConstants.THING_TYPE_MOON, LOCAL);
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (astroDiscoveryJob == null) {
+            astroDiscoveryJob = scheduler.scheduleAtFixedRate(() -> {
+                PointType currentLocation = locationProvider.getLocation();
+                if ((currentLocation != null) && !currentLocation.equals(previousLocation)) {
+                    logger.info("Location has been changed from {} to {}: Creating new Discovery Results",
+                            previousLocation, currentLocation);
+                    createResults(currentLocation);
+                    previousLocation = currentLocation;
+                }
+            }, 0, LOCATION_CHANGED_CHECK_INTERVAL, TimeUnit.SECONDS);
+            logger.debug("Scheduled astro location-changed job every {} seconds", LOCATION_CHANGED_CHECK_INTERVAL);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stopping Astro device background discovery");
+        if (astroDiscoveryJob != null && !astroDiscoveryJob.isCancelled()) {
+            astroDiscoveryJob.cancel(true);
```

Comment:
you can actually use the returned boolean value to nullify `astroDiscoveryJob` and print the log.